src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -203,5 +238,43 @@ private BoundNode CollectDynamicAnalysis(BoundStatement statement)
             
             return statementFactory.Block(ImmutableArray.Create(cellAssignment, statement));
         }
+
+        private static CSharpSyntaxNode SyntaxForSpan(BoundStatement statement)
```

Comment:
It feels like we are trying to reinvent  something that already exists. It looks to me that spans that we assign for instrumentation should be similar to spans that we assign for sequence points. And to do this properly, we need to integrate instrumentation into the lowering pass. Let's not invest any more into the current approach beyond this change. Let's move to the design we would like to have first.